File: 0342-power-of-four/0342-power-of-four.py
class Solution:
    def isPowerOfFour(self, n: int) -> bool:
# negative and zero cannot be a pow() of any number
        if n <= 0:
            return False
# when number is greater than zero and if number becomes 1 on continuous division without leaving out any remainder will the pow() of any number
        while n > 0:
            if n == 1:
                return True
            elif n%4 != 0:
                break
            n //= 4
        return False

File: 0342-power-of-four/0342-power-of-four.java
class Solution {
    public boolean isPowerOfFour(int n) {
        // while n > 0
        while(n>0){
            // if n becomes 1 then n is power of 4
            if (n==1)   return true;
            // if remainder on dividing by 4 is not 1 then not a pow of 4
            else if (n%4 != 0)  break;
            // divide n by 4
            n /= 4;
        }
        // return false if not pow of 4
        return false; 
    }
}